modules/network/networker.go

```diff
@@ -65,12 +70,98 @@ func validateNetwork(n *modules.Network) error {
 	return nil
 }
 
-func (n networker) Namespace(id modules.NetID) (string, error) {
-	b, err := ioutil.ReadFile(filepath.Join(n.storageDir, string(id)))
+func (n networker) namespaceOf(net *modules.Network) (string, error) {
+	nibble, err := n.nibble(net)
+	if err != nil {
+		return "", err
+	}
+	return nibble.NetworkName(), nil
+}
+
+func (n networker) bridgeOf(net *modules.Network) (string, error) {
+	nibble, err := n.nibble(net)
 	if err != nil {
 		return "", err
 	}
-	return string(b), err
+	return nibble.BridgeName(), nil
+}
+
+func (n *networker) Join(id modules.NetID) (name string, err error) {
+	// TODO:
+	// 1- Make sure this network id is actually deployed
+	// 2- Create a new namespace, then create a veth pair inside this namespace
+	// 3- Hook one end to the NR bridge
+	// 4- Assign IP to the veth endpoint inside the namespace.
+	// 5- return the namespace name
+
+	log.Info().Str("network-id", string(id)).Msg("joining network")
+
+	net, err := n.networkOf(id)
+	if err != nil {
+		return "", errors.Wrapf(err, "couldn't load network with id (%s)", id)
+	}
+	// 1- Make sure this network is is deployed
+	brName, err := n.bridgeOf(net)
+	if err != nil {
+		return name, errors.Wrapf(err, "failed to get bridge for network: %v", id)
+	}
+
+	br, err := bridge.Get(brName)
+	if err != nil {
+		return name, err
+	}
+
+	// TODO: the number in the name below must be changed to a deterministic
+	// value, may be driven from the IP assigned to this namespace
+	name = fmt.Sprintf("%s-%d", "container", 1) //TODO: CHANGE ME
```

Comment:
The name here for the veth pair i have no idea yet what it should be, but may be part of the ip or the container name, may be thename should be passed to the join?